src/__tests__/EndtoEnd.test.ts

```diff
@@ -0,0 +1,103 @@
+import puppeteer from 'puppeteer';
+import { mockData } from '../mock-data';
+import { onlyUnique } from '../utils';
+
+describe('filter events by city', () => {
+  let browser: puppeteer.Browser;
+  let page: puppeteer.Page;
+  let suggestions: puppeteer.ElementHandle<Element>[];
+
+  beforeAll(async () => {
+    browser = await puppeteer.launch();
+    page = await browser.newPage();
+    await page.goto('http://localhost:3000/');
+    await page.waitForSelector('.event');
+  });
+  afterAll(() => browser.close());
+  test('When user hasn’t searched for a city, show upcoming events from all cities', async () => {
+    const eventlocationElements = await page.$$('.event_location');
+    expect(eventlocationElements).toHaveLength(mockData.length);
+    const actualLocations = await Promise.all(
+      eventlocationElements.map(async (element) => {
+        const titleAndLocation = await (
+          await element.getProperty('textContent')
+        ).jsonValue();
+        const location = titleAndLocation?.split('| ')[1];
+        return location;
+      }),
+    );
+    // actual events are set in different locations
```

Comment:
Although commenting code is important, a lot or too much explanations can make things much hard to navigate through. 

Try to concise and verberlly